tool.go

```diff
@@ -121,13 +121,22 @@ func main() {
 
 				for _, pkgPath := range args {
 					pkgPath = filepath.ToSlash(pkgPath)
-					if s.Watcher != nil {
-						s.Watcher.Add(pkgPath)
-					}
 					pkg, err := gbuild.Import(pkgPath, 0, s.InstallSuffix(), options.BuildTags)
 					if err != nil {
 						return err
 					}
+					if s.Watcher != nil {
+						s.Watcher.Add(pkg.Dir)
+
+						// We want to have the package dir watched when calling Import to avoid
+						// races where we could miss a change made and not re-trigger a build.
```

Comment:
Just want to make sure I understand this part.

Is the reason for the above because _calling Import may take a while_, so if a file is changed there while `gbuild.Import` is called, you want to be able to detect that. Or is there another reason?

If that's true, then why is it okay to not have package dir watched when the "first" `gbuild.Import` runs. Surely, it would take as long.